### ISSUE TITLE
Consider adding a base `forc tx` command for constructing arbitrary transactions from the CLI

### ISSUE THREAD

**Issue by mitchmindtree** [FuelLabs/sway#3491]
Currently `forc-client` provides:

- `forc deploy` for constructing `Create` transactions (for deploying contracts).
- `forc run` for constructing `Script` transactions (for executing scripts).

Conceptually these cover all main types of transactions, however both are currently very limited compared to constructing `Create` or `Script` transactions using one of the SDKs directly.

I wonder if it might be useful to take a step back, and think about how we can provide a nice CLI experience for constructing arbitrary transactions in general, perhaps via a `forc tx` command.

## Constructing transactions (`forc tx`)

Here is a _very rough_ example of how an invocation might look:

```console
# Create a `Create` type transaction.
forc tx create \
    --bytecode ./out/debug/my-contract.bin \
    --storage-slots ./out/debug/my-contract-storage-slots.json \
    --salt ... \
    --gas-price 1 \
    --gas-limit 10 \
    --maturity 1 \
    --inputs "./inputs.json" \
    --outputs "./outputs.json" \
    --witness ... \
    --witness ...
```

I'm glossing over significant details like `inputs.json` and `outputs.json` here, but the general idea is to provide a more comprehensive mapping to whats possible in the spec (and in turn the Rust SDK).

We could limit the role of `forc tx` to actually *constructing* the transaction. By default, it could pretty print the result in JSON to stdout, or accept a flag for writing it to a file.

Arguments like `--bytecode` and `--storage-slots` could be omitted and inferred if running the command within a forc project, as there are standard locations within the project where we can fetch these values (i.e. `./out/<profile>/...`).

## Submitting transactions (`forc submit`)

We could then use another command like `forc submit my-transaction.json` to submit the full transaction to a network, allowing to specify a custom URL and other network-related args.

## `forc deploy`, `forc run`

Commands like `forc deploy` and `forc run` would then become thin, higher-level wrappers around a combination of `forc tx create`/`forc tx script` and `forc submit`.

## Why care about exposing this from a CLI?

Benefits of having a purely CLI approach include:

- Just about every OS comes with Bash out of the box.
- No need for Sway devs to install or learn potentially unfamiliar languages (like Rust or TS).
- Easier to write custom deployment/execution scripts without the need to build/compile them.
- Be more Unix friendly.

**Comment by nfurfaro** [comment]
I'm really liking the direction this is going!

If possible, it would be really cool if the final transaction could (in addition to being printed to stdout/written to file) be output as the corresponding raw bytecode. This would allow us to do things like `eth_sendRawTransaction` and "keyless transactions"("[Nick's method](https://weka.medium.com/how-to-send-ether-to-11-440-people-187e332566b7)")

**Comment by mitchmindtree** [comment]
Closed via #3582, though improvements around signing and bug fixes (i.e. #3886) to come.
